### PR TITLE
Update to dehydrated 0.7.1 to fix LE staging CA and various improvements

### LINKED ISSUE #63: Want dehydrated version 0.7.1 to fix LE staging usage
When trying to get a certificate via the Let's Encrypt staging CA, the following error comes up in the log after a valid challenge:
```
ERROR: Problem connecting to server (post for ; curl returned with 3)
```

From https://github.com/dehydrated-io/dehydrated/releases/tag/v0.7.1
> Requests resulting in badNonce errors are now automatically retried (fixes operation with LE staging servers)

### PR DESCRIPTION
**Pull request by teutat3s**
Fixes #63.

Release notes:
https://github.com/dehydrated-io/dehydrated/releases/tag/v0.7.1

Removed patch that was added in #54, the fix has been integrated upstream in https://github.com/dehydrated-io/dehydrated/pull/868 and https://github.com/dehydrated-io/dehydrated/pull/840.

### CODE REVIEW

File: cns-hook-util [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 # Copyright 2021 Joyent, Inc.
 # Copyright 2023 MNX Cloud, Inc.
```
Comment (bahamat) [inline cns-hook-util]: MNX only needs one copyright line with the most recent year.

```
# Copyright 2021 Joyent, Inc.
# Copyright 2024 MNX Cloud, Inc.

```

File: Makefile
Comment (bahamat) [inline Makefile:1]: This file needs copyright update:

```
# Copyright 2021 Joyent, Inc.
# Copyright 2024 MNX Cloud, Inc.

```

**Review by bahamat — CHANGES_REQUESTED** [review]

File: cns-hook-util (hunk 1 above)
Comment (teutat3s) [reply cns-hook-util]: Done in 9813e31a2b1cff3cb07994bde4b5270ee6c8aaab.

File: Makefile
Comment (teutat3s) [reply Makefile:1]: Done in 9813e31a2b1cff3cb07994bde4b5270ee6c8aaab.

**Review by bahamat — APPROVED** [review]

**Comment by bahamat** [comment]
This has now been tagged and released as v1.5.3.

**Pull request merged**